### PR TITLE
Revert (most of) "Makefile: Generate crypto objects only as far as needed"

### LINKED ISSUE #19087: Build errors for buildtests in parallel builds due to missing dependency
When using parallel make with enabled buildtests, we sometimes see failures like

```
In file included from test/buildtest_obj_mac.c:10:
include/openssl/obj_mac.h:12: error: unterminated #ifndef
  12 | #ifndef OPENSSL_OBJ_MAC_H
```

This happens because `test/buildtest_obj_mac.c` does not have an explicit dependency on `include/openssl/obj_mac.h` which is regenerated in parallel due to the rules added in commit d040a1b9a028e89f0a33b36fb99f0151d2fdd4c3.  The rebuild path in the makefile is:

```
build_sw
->
_build_sw
->
build_libs_nodep
->
libcrypto.pc
->
libcrypto.so
->
libcrypto.so.3
->
libcrypto.ld
->
util/libcrypto.num
->
include/openssl/obj_mac.h (via CRYPTOHEADERS)
```

Since there is no dependency between this chain and the compilation of `build_test_obj_mac.c` into `test/buildtest_c_obj_mac-bin-buildtest_obj_mac.o`, they can run in parallel.  In rare cases, this leads to the situation where regeneration of obj_mac.h is still taking place while compilation reads that (partial) file causing above mentioned error.

I am not sure how to fix this issue.  Once solution would be to add an explicit dependency between the compilation step and the generated header.

Note that this problem does not occur always since parallel builds are non-deterministic.

### PR DESCRIPTION
**Pull request by DDvO**
This reverts the adaptations to ~`generate_crypto_objects`~ `Configurations/unix-Makefile.tmpl` (except for additions to `.PHONY`) that were done by commit 89c39bd8f6528ad268c1c0d0c9e196ba4396941b in #15224.

This ~is supposed to fix~ most likely fixes #19087.

### CODE REVIEW

**Review by K-kingp — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Ping @openssl/committers for 2nd approval -
this is somewhat urgent because it fixes occasional build hick-ups.

**Comment by DDvO** [comment]
CI failure is unrelated.

**Comment by DDvO** [comment]
> What makes you think that the problem found in #19087 can't happen with the other header files that were moved around by #15224?

Well, I hope that it does not happen with them, but the only indication that I have in this direction is that I am not aware of any issue reported in this regard.

**Comment by levitte** [comment]
Unfortunately, we found new cases where having a dependency based `make update` doesn't get the job done right.  This time, it has to do with timestamps that a git checkout produces.  See https://github.com/openssl/openssl/pull/19197#issuecomment-1243965112

**Comment by DDvO** [comment]
To be on the safe side, 
I've extended this PR to revert all adaptations by commit 89c39bd8f6528ad268c1c0d0c9e196ba4396941b
on `Configurations/unix-Makefile.tmpl` except for additions to `.PHONY`.

**Comment by t8m** [comment]
Would you please add as a separate commit the update of openssl-vms.cnf that the check-update CI build reports? So the CI would be green after this is merged?

**Comment by DDvO** [comment]
Done.
Shall we mark this PR urgent?

**Review by levitte — APPROVED** [review]

**Comment by levitte** [comment]
I've labeled this urgent.

@t8m, re-approve?

**Comment by levitte** [comment]
Oh, a nit: drop "(most of)" from the commit message, since you have now reverted all of it

**Review by t8m — APPROVED** [review]
OK with urgent

**Comment by DDvO** [comment]
Merged - thanks @levitte and @t8m for the swift handling.

**Pull request closed without merging**
